### ISSUE TITLE
Typescript SDK seems to drop parameters for queries

### ISSUE THREAD

**Issue by ADNolan** [contentstack/contentstack-typescript#96]
Version: `@contentstack/delivery-sdk@4.3.0`

Sample query:

```
stack
   .contentType('my_content_type')
  .entry()
  .locale('en-us')
  .includeFallback();
  .includeReference([
      'valid.reference.field1',
      'valid.reference.field2',
    ])
  .includeCount()
  .limit(1)
```

Logging some output, the sample is

```
 Querying ContentStack with {"_queryParams":{"locale":"en-us","include_fallback":"true","include[]":["valid.reference.field1","valid.reference.field2"],"include_count":"true","limit":1},"_parameters":{},"_contentTypeUid":"my_content_type","_urlPath":"/content_types/my_content_type/entries"}
```

However, the shape of the `FindResponse` doesn't include `count` and all the entries are returned instead of just 1. When using other methods like `equalTo`, `where` and the like, those are seemingly not applied to filter results further.

Trying a request via postman that I believe should equivalent, I get the expected results. This one has an added filter for a specific entry for an exaple.

```
https://{{base_url}}/v3/content_types/my_content_type/entries?locale=en-us&include_fallback=true&include_branch=false&query={"uid":"bltdd*******"}&include[]=valid.reference.field1&include[]=valid.reference.file2&environment=qa&include_count=true&limit=1
```

One reason I have gone down this path is that `includeReferences` is not able to be applied to the `Entry` type. The workaround was to use the `Entries` type and filter it down. Base on the SDK docs, it seems like this should be possible to include references on the Entries query builder, but it errors out with method is not defined error.

Here is the logged out Axios Response for a real sample instead of using replaced names.

```
{
  status: 200,
  statusText: 'OK',
  headers: Object [AxiosHeaders] {
    connection: 'close',
    'content-length': '38049',
    'content-type': 'application/json',
    'x-cs-includecount': '288',
    'x-cs-includedepth': '1',
    'x-contentstack-organization': '***',
    'x-runtime': '477ms',
    'x-request-id': 'a36b66c1-6e63-47f2-8e30-02566b4d8c33',
    server: 'contentstack',
    via: '1.1 varnish, 1.1 varnish',
    'accept-ranges': 'bytes',
    age: '1750',
    date: 'Wed, 11 Dec 2024 22:28:39 GMT',
    'x-served-by': 'cache-bfi-kbfi7400044-BFI, cache-bur-kbur8200032-BUR',
    'x-cache': 'HIT, HIT',
    'x-cache-hits': '1, 0',
    'x-timer': 'S1733956120.937631,VS0,VE1',
    vary: 'Accept-Encoding',
    'access-control-allow-origin': '*',
    'strict-transport-security': 'max-age=900'
  },
  config: {
    transitional: {
      silentJSONParsing: true,
      forcedJSONParsing: true,
      clarifyTimeoutError: false
    },
    adapter: [ 'xhr', 'http', 'fetch' ],
    transformRequest: [ [Function: transformRequest] ],
    transformResponse: [ [Function: transformResponse] ],
    timeout: 30000,
    xsrfCookieName: 'XSRF-TOKEN',
    xsrfHeaderName: 'X-XSRF-TOKEN',
    maxContentLength: -1,
    maxBodyLength: -1,
    env: { FormData: [Function], Blob: [class Blob] },
    validateStatus: [Function: validateStatus],
    headers: Object [AxiosHeaders] {
      Accept: 'application/json, text/plain, */*',
      'Content-Type': undefined,
      api_key: '***',
      access_token: '***',
      'X-User-Agent': 'contentstack-delivery-typescript-{{PLATFORM}}/{{VERSION}}',
      'User-Agent': 'axios/1.7.9',
      'Accept-Encoding': 'gzip, compress, deflate, br'
    },
    baseURL: 'https://cdn.***-contentstack.com:443/v3',
    insecure: false,
    retryOnError: true,
    basePath: '',
    proxy: false,
    httpAgent: false,
    httpsAgent: false,
    logHandler: [Function: logHandler],
    retryCondition: [Function: retryCondition],
    versioningStrategy: 'path',
    defaultHostname: 'cdn.***-contentstack.com',
    params: { environment: 'qa', query: undefined },
    live_preview: {},
    paramsSerializer: { serialize: [Function: serialize] },
    locale: 'en-us',
    include_fallback: 'true',
    'include[]': [
      'media_gallery_section.items_v2.heading',
      'media_gallery_section.items_v2.media'
    ],
    include_count: 'true',
    limit: 1,
    query: { uid: 'blt***' },
    method: 'get',
    url: '/content_types/home_page/entries',
    retryCount: 1,
    data: undefined
  },
  request: <ref *1> ClientRequest {
    _events: [Object: null prototype] {
      abort: [Function (anonymous)],
      aborted: [Function (anonymous)],
      connect: [Function (anonymous)],
      error: [Function (anonymous)],
      socket: [Function (anonymous)],
      timeout: [Function (anonymous)],
      finish: [Function: requestOnFinish]
    },
    _eventsCount: 7,
    _maxListeners: undefined,
    outputData: [],
    outputSize: 0,
    writable: true,
    destroyed: true,
    _last: true,
    chunkedEncoding: false,
    shouldKeepAlive: false,
    maxRequestsOnConnectionReached: false,
    _defaultKeepAlive: true,
    useChunkedEncodingByDefault: false,
    sendDate: false,
    _removedConnection: false,
    _removedContLen: false,
    _removedTE: false,
    strictContentLength: false,
    _contentLength: 0,
    _hasBody: true,
    _trailer: '',
    finished: true,
    _headerSent: true,
    _closed: true,
    socket: TLSSocket {
      _tlsOptions: [Object],
      _secureEstablished: true,
      _securePending: false,
      _newSessionPending: false,
      _controlReleased: true,
      secureConnecting: false,
      _SNICallback: null,
      servername: 'cdn.***-contentstack.com',
      alpnProtocol: false,
      authorized: true,
      authorizationError: null,
      encrypted: true,
      _events: [Object: null prototype],
      _eventsCount: 9,
      connecting: false,
      _hadError: false,
      _parent: null,
      _host: 'cdn.***-contentstack.com',
      _closeAfterHandlingError: false,
      _readableState: [ReadableState],
      _writableState: [WritableState],
      allowHalfOpen: false,
      _maxListeners: undefined,
      _sockname: null,
      _pendingData: null,
      _pendingEncoding: '',
      server: undefined,
      _server: null,
      ssl: null,
      _requestCert: true,
      _rejectUnauthorized: true,
      parser: null,
      _httpMessage: [Circular *1],
      timeout: 30000,
      autoSelectFamilyAttemptedAddresses: [Array],
      write: [Function: writeAfterFIN],
      [Symbol(alpncallback)]: null,
      [Symbol(res)]: null,
      [Symbol(verified)]: true,
      [Symbol(pendingSession)]: null,
      [Symbol(async_id_symbol)]: 330274,
      [Symbol(kHandle)]: null,
      [Symbol(lastWriteQueueSize)]: 0,
      [Symbol(timeout)]: Timeout {
        _idleTimeout: -1,
        _idlePrev: null,
        _idleNext: null,
        _idleStart: 19534,
        _onTimeout: null,
        _timerArgs: undefined,
        _repeat: null,
        _destroyed: true,
        [Symbol(refed)]: false,
        [Symbol(kHasPrimitive)]: false,
        [Symbol(asyncId)]: 330375,
        [Symbol(triggerId)]: 330277,
        [Symbol(kResourceStore)]: undefined,
        [Symbol(kResourceStore)]: undefined,
        [Symbol(kResourceStore)]: [Object],
        [Symbol(kResourceStore)]: [Object],
        [Symbol(kResourceStore)]: [Object],
        [Symbol(kResourceStore)]: undefined
      },
      [Symbol(kBuffer)]: null,
      [Symbol(kBufferCb)]: null,
      [Symbol(kBufferGen)]: null,
      [Symbol(shapeMode)]: true,
      [Symbol(kCapture)]: false,
      [Symbol(kSetNoDelay)]: false,
      [Symbol(kSetKeepAlive)]: true,
      [Symbol(kSetKeepAliveInitialDelay)]: 60,
      [Symbol(kBytesRead)]: 38693,
      [Symbol(kBytesWritten)]: 377,
      [Symbol(connect-options)]: [Object]
    },
    _header: 'GET /v3/content_types/home_page/entries?environment=qa HTTP/1.1\r\n' +
      'Accept: application/json, text/plain, */*\r\n' +
      'api_key: ***\r\n' +
      'access_token: ***\r\n' +
      'X-User-Agent: contentstack-delivery-typescript-{{PLATFORM}}/{{VERSION}}\r\n' +
      'User-Agent: axios/1.7.9\r\n' +
      'Accept-Encoding: gzip, compress, deflate, br\r\n' +
      'Host: cdn.***-contentstack.com\r\n' +
      'Connection: close\r\n' +
      '\r\n',
    _keepAliveTimeout: 0,
    _onPendingData: [Function: nop],
    agent: Agent {
      _events: [Object: null prototype],
      _eventsCount: 2,
      _maxListeners: undefined,
      defaultPort: 443,
      protocol: 'https:',
      options: [Object: null prototype],
      requests: [Object: null prototype] {},
      sockets: [Object: null prototype] {},
      freeSockets: [Object: null prototype] {},
      keepAliveMsecs: 1000,
      keepAlive: false,
      maxSockets: Infinity,
      maxFreeSockets: 256,
      scheduling: 'lifo',
      maxTotalSockets: Infinity,
      totalSocketCount: 0,
      maxCachedSessions: 100,
      _sessionCache: [Object],
      [Symbol(shapeMode)]: false,
      [Symbol(kCapture)]: false
    },
    socketPath: undefined,
    method: 'GET',
    maxHeaderSize: undefined,
    insecureHTTPParser: undefined,
    joinDuplicateHeaders: undefined,
    path: '/v3/content_types/home_page/entries?environment=qa',
    _ended: true,
    res: IncomingMessage {
      _events: [Object],
      _readableState: [ReadableState],
      _maxListeners: undefined,
      socket: [TLSSocket],
      httpVersionMajor: 1,
      httpVersionMinor: 1,
      httpVersion: '1.1',
      complete: true,
      rawHeaders: [Array],
      rawTrailers: [],
      joinDuplicateHeaders: undefined,
      aborted: false,
      upgrade: false,
      url: '',
      method: null,
      statusCode: 200,
      statusMessage: 'OK',
      client: [TLSSocket],
      _consuming: true,
      _dumped: false,
      req: [Circular *1],
      _eventsCount: 4,
      responseUrl: 'https://cdn.***-contentstack.com/v3/content_types/home_page/entries?environment=qa',
      redirects: [],
      [Symbol(shapeMode)]: true,
      [Symbol(kCapture)]: false,
      [Symbol(kHeaders)]: [Object],
      [Symbol(kHeadersCount)]: 42,
      [Symbol(kTrailers)]: null,
      [Symbol(kTrailersCount)]: 0
    },
    aborted: false,
    timeoutCb: null,
    upgradeOrConnect: false,
    parser: null,
    maxHeadersCount: null,
    reusedSocket: false,
    host: 'cdn.***-contentstack.com',
    protocol: 'https:',
    _redirectable: Writable {
      _events: [Object],
      _writableState: [WritableState],
      _maxListeners: undefined,
      _options: [Object],
      _ended: true,
      _ending: true,
      _redirectCount: 0,
      _redirects: [],
      _requestBodyLength: 0,
      _requestBodyBuffers: [],
      _eventsCount: 3,
      _onNativeResponse: [Function (anonymous)],
      _currentRequest: [Circular *1],
      _currentUrl: 'https://cdn.***-contentstack.com/v3/content_types/home_page/entries?environment=qa',
      _timeout: null,
      [Symbol(shapeMode)]: true,
      [Symbol(kCapture)]: false
    },
    [Symbol(shapeMode)]: false,
    [Symbol(kCapture)]: false,
    [Symbol(kBytesWritten)]: 0,
    [Symbol(kNeedDrain)]: false,
    [Symbol(corked)]: 0,
    [Symbol(kOutHeaders)]: [Object: null prototype] {
      accept: [Array],
      api_key: [Array],
      access_token: [Array],
      'x-user-agent': [Array],
      'user-agent': [Array],
      'accept-encoding': [Array],
      host: [Array]
    },
    [Symbol(errored)]: null,
    [Symbol(kHighWaterMark)]: 16384,
    [Symbol(kRejectNonStandardBodyWrites)]: false,
    [Symbol(kUniqueHeaders)]: null
  },
  data: {
    entries: [ [Object], [Object], [Object], [Object], [Object], [Object] ]
  }
}
```

Bit hazy on debugging Axios, but it seems the request with just called with `https://cdn.***-contentstack.com/v3/content_types/home_page/entries?environment=qa` Which is completely missing every other param that should be set. Looking at the response instance, the properties I would expect to be sent as part of the real request are present.
```
locale: 'en-us',
    include_fallback: 'true',
    'include[]': [
      'media_gallery_section.items_v2.heading',
      'media_gallery_section.items_v2.media'
    ],
    include_count: 'true',
    limit: 1,
    query: { uid: 'blt***' },
    method: 'get',
    url: '/content_types/home_page/entries',
    retryCount: 1,
    data: undefined
  },
```

**Comment by ADNolan** [comment]
Adding some additional findings. I found that our node package that wraps the content delivery SDK that was working standalone in Unit / Functional tests because of the locally installed version `@contentstack/core` needed for the delivery sdk.

The module integrating/wrapping the delivery SDK had  `@contentstack/core@1.1.1` installed locally. The params for the request were applied appropriately.

When installing our wrapper npm package in a test application, `@contentstack/core@1.1.3` was installed and that is where things go wrong. This version was installed based on the `"@contentstack/core": "^1.1.0",` in the delivery sdk package json.

 I verified this a by adjusting versioning in the local package json for `@contentstack/delivery-sdk` to explicitly install  `@contentstack/core@1.1.1` and start seeing expected result and the request/response is applying query params.

**Comment by nadeem-cs** [comment]
Hey @ADNolan , 

Thanks for raising this!

We’re aware of this issue, and I’m happy to share that it was resolved in version 4.4.0 of the package. 

The latest version is now v4.4.3. We recommend updating to the latest version for the best experience.

Let us know if you face any issues after upgrading!
